apidef/oas/operation.go

```diff
@@ -0,0 +1,157 @@
+package oas
+
+import (
+	"strings"
+
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+type Operations map[string]*Operation
+
+type Operation struct {
+	Allow                *Allowance `bson:"allow,omitempty" json:"allow,omitempty"`
+	Block                *Allowance `bson:"block,omitempty" json:"block,omitempty"`
+	IgnoreAuthentication *Allowance `bson:"ignoreAuthentication,omitempty" json:"ignoreAuthentication,omitempty"`
+}
+
+const (
+	allow                AllowanceType = 0
+	block                AllowanceType = 1
+	ignoreAuthentication AllowanceType = 2
+)
+
+type AllowanceType int
+
+func (s *OAS) fillPathsAndOperations(ep apidef.ExtendedPathsSet) {
+	if s.Paths == nil {
+		s.Paths = make(openapi3.Paths)
+	}
+
+	s.fillAllowance(ep.WhiteList, allow)
+	s.fillAllowance(ep.BlackList, block)
+	s.fillAllowance(ep.Ignored, ignoreAuthentication)
+
+	if len(s.Paths) == 0 {
+		s.Paths = nil
+	}
+}
+
+func (s *OAS) extractPathsAndOperations(ep *apidef.ExtendedPathsSet) {
+	tykOperations := s.getTykOperations()
+	if len(tykOperations) == 0 {
+		return
+	}
+
+	for id, tykOp := range tykOperations {
+		for path, pathItem := range s.Paths {
+			for method, operation := range pathItem.Operations() {
+				if id == operation.OperationID {
+					tykOp.extractAllowanceTo(ep, path, method, allow)
+					tykOp.extractAllowanceTo(ep, path, method, block)
+					tykOp.extractAllowanceTo(ep, path, method, ignoreAuthentication)
+				}
+			}
+		}
+	}
+}
+
+func (s *OAS) fillAllowance(endpointMetas []apidef.EndPointMeta, typ AllowanceType) {
+	for _, em := range endpointMetas {
+		operationID := s.getOperationID(em.Path, em.Method)
+		operation := s.GetTykExtension().getOperation(operationID)
+
+		var allowance *Allowance
```

Comment:
don't we have to assign allowance to OAS object field?

Reply:
As it is a pointer we set it like this for example `allowance = operation.Block` and when you change it, `operation.Block` changes.